### PR TITLE
Several changes for 3D Tiles api 

### PR DESCRIPTION
**Pull request by bertt**
. IMeshFeatureIDInfo, FeatureIdBuilder moved to new file

. Renamed method SetValues1D -> SetValues (because most frequently used)

- Renamed method SetValues2D -> SetArrayValues

. Changed FeatureIDBuilder constructor to 

```
FeatureIDBuilder(int featureCount, OneOf<int, Texture>? attributeOrTexture = null, PropertyTable propertyTable = null, IReadOnlyList<int> channels = null, string label = null, int? nullFeatureId = null)
```
. Changed method AddMeshFeatureIds(this MeshPrimitive primitive, params (IMeshFeatureIDInfo fid, Texture tex, IReadOnlyList<int> channels)[] featureIds)

to AddMeshFeatureIds(this MeshPrimitive primitive, params IMeshFeatureIDInfo[] fids)

.  Improve support for array types (determine automatically if the array is fixed length or not)

. added type checking of metadata  values to the schema

. fixed enums (must use short type)

. Tested all Cesium tests https://github.com/CesiumGS/3d-tiles-validator/tree/main/specs/data/gltfExtensions

. Added README.md with sample code

NOT supported now:

. external schema (using a json file) support; 

. reading/writing/validating properties min, max, offset and scale.

### CODE REVIEW

File: src/SharpGLTF.Ext.3DTiles/Schema2/Ext.Features.cs [hunk 1]
```diff
@@ -235,14 +235,6 @@ public MeshExtMeshFeatureID CreateFeatureID(IMeshFeatureIDInfo properties, Textu
                 instance.Label = properties.Label;
                 instance.Attribute = properties.Attribute;
                 instance.PropertyTableIndex = properties.PropertyTableIndex;
-
-                if (texture != null)
```
Comment (bertt) [inline src/SharpGLTF.Ext.3DTiles/Schema2/Ext.Features.cs:239]: must be added somewhere else?

Comment (bertt) [reply src/SharpGLTF.Ext.3DTiles/Schema2/Ext.Features.cs:239]: added again https://github.com/vpenades/SharpGLTF/pull/213/commits/93f3d5178e053d5962954ab99f5cc46d26085386

File: src/SharpGLTF.Ext.3DTiles/Schema2/Ext.StructuralMetadataRoot.cs [hunk 2]
```diff
@@ -95,9 +95,16 @@ public StructuralMetadataSchema UseEmbeddedSchema(string id)
                 return schema;
             }
 
+            // Sets the schema to use an external schema, returns an empty schema to used for adding schema properties
+            public StructuralMetadataSchema UseExternalSchema(Uri uri)
```
Comment (bertt) [inline src/SharpGLTF.Ext.3DTiles/Schema2/Ext.StructuralMetadataRoot.cs:99]: here the method for setting external schema uri, it works but doesn't use a ISchemaProxy

File: src/SharpGLTF.Core/Schema2/_Extensions.Nullables.cs [hunk 3]
```diff
@@ -1,7 +1,5 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Numerics;
-using System.Text;
```
Comment (vpenades) [inline src/SharpGLTF.Core/Schema2/_Extensions.Nullables.cs:4]: Although these changes are fine, they're not relevant to this PR.

**Review by vpenades — CHANGES_REQUESTED** [review]
Other that that, everything else looks fine.

File: src/SharpGLTF.Ext.3DTiles/Schema2/Ext.StructuralMetadataRoot.cs (hunk 2 above)
Comment (bertt) [reply src/SharpGLTF.Ext.3DTiles/Schema2/Ext.StructuralMetadataRoot.cs:103]: external schema is not supported for now

**Pull request merged**